### ISSUE TITLE
Coco128 dataset -- labels do not match images

### ISSUE THREAD

**Issue by jrivas93** [ultralytics/ultralytics#5342]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

File: `ultralytics/ultralytics/data/utils.py`
Function: `check_det_dataset()`
Line: 290

**Issue:** When calling model.train() with the base coco128.yml dataset, the code first checks the `train` and `val` paths in the YAML to see if there are valid images and labels. If not, it uses the `download` field in the coco128.yaml file to download the dataset from a specified URL. In the case of the coco128.yaml, the field is populated as `download: https://ultralytics.com/assets/coco128.zip`

Upon inspection of this dataset, it appears that although both the images and labels directories both contain 128 images, the images are not correctly aligned with their respective label file. For example, it is expected that image `000000000009.jpg` has a corresponding label `000000000009.txt`. As an example of the mismatched images and labels, the labels directory contains files `000000000656.txt` and `000000000659.txt` ; their corresponding images are not present in the images directory. The last image in the images directory is `000000000650.jpg`, and is of a cat. Inspection of the `000000000650.txt` confirms that class 15 is present in this image, and class 15 is labeled as a cat.

**Potential Fix**: Update the 128-image subset of the entire 5000 image COCO dataset so that all 128 images are correctly paired with their labels. 

**Importance:** It is important to have correctly labeled data for a computer vision task. It is possible that some users may be experiencing higher than expected error rates due to this mismatch.

**Understanding & Clarification of Model:** How does the yolov8 code handle the pairing of images and labels? If we have a label `000000000659.txt` but do not have an image `000000000659.jpg` is this instance ignored during training? If so, then the above issue does not matter as much, since yolov8 is only training using images that have a corresponding label. Thus the only effect this has is that the training dataset is not a full 128 images, but rather only the subset of images that have a matching label.

### Environment

Ultralytics YOLOv8.0.196 🚀 Python-3.9.12 torch-2.1.0+cu121 CPU (11th Gen Intel Core(TM) i7-1165G7 2.80GHz)

### Minimal Reproducible Example

``` 
from ultralytics import YOLO

# Load a model
model = YOLO("yolov8n.pt")  # load a pretrained model (recommended for training)

# Use the model
model.train(data="coco128.yaml", epochs=3)  # train the model
```

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@jrivas93 hello, and thank you for the detailed report! 

We appreciate the extensive investigation that helps us understand the issue you're facing with mismatched images and labels in the `coco128` dataset. We understand how crucial it is to have correctly labeled data for any computer vision task and we'll work to rectify this issue.

To answer your question about how YOLOv8 handles the pairing of images and labels, indeed, images without corresponding labels and vice-versa are ignored during training. So, the subset of images used indeed includes only those that have matching labels. 

While that mitigates some impact of the mismatch, we recognize the need for clarity and consistency in the dataset. We'll update the 128-image subset of the COCO dataset to ensure all images are correctly paired with their labels. 

Thanks for offering to contribute via a PR, it's that sort of community collaboration that helps keep projects like this successful and continually improving. You're absolutely right about the importance of accurate and complete datasets for training and we're grateful for users like you who help to enhance our service. 

Do let us know if you have any other questions or concerns. Happy training!
